### PR TITLE
Adds check to avoid usage of globals

### PR DESCRIPTION
**Pull request by micheleorselli**
Hey!

Type: new feature

Link to issue:

This pull request affects the following components **(please check boxes):**

* [ ] Core
* [x] Analyzer
* [ ] Compiler
* [ ] Control Flow Graph
* [ ] Documentation

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/ovr/phpsa/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Implements #213. I'd like to have feedback particularly on the test code, just to be sure I got it right, thanks!

### CODE REVIEW

File: tests/analyze-fixtures/Statement/DoNotUseGlobals.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+        return $a * 3;
+    }
+
+}
+
+?>
+----------------------------
+[{
+    "type": "missing_docblock",
```
Comment (micheleorselli) [inline tests/analyze-fixtures/Statement/DoNotUseGlobals.php]: not sure why I'm getting these errors, should be the missing global check the one triggered shouldn't be?

File: tests/analyze-fixtures/Statement/DoNotUseGlobals.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    "message": "Missing Docblock",
+    "file": "DoNotUseGlobals.php",
+    "line": 6
+}, {
+    "type": "do_not_use_globals",
+    "message": "Do not use globals",
+    "file": "DoNotUseGlobals.php",
+    "line": 13
```
Comment (ddmler) [inline tests/analyze-fixtures/Statement/DoNotUseGlobals.php]: you only need this one of type do_not_use_globals. you have to remove the other ones.

Comment (micheleorselli) [inline tests/analyze-fixtures/Statement/DoNotUseGlobals.php]: ok, any hints on how to do that?

File: tests/analyze-fixtures/Statement/DoNotUseGlobals.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Tests\Analyze\Fixtures\Statement;
+
+$a = 1;
+
+class DoNotUseGlobals {
```
Comment (sergeyklay) [inline tests/analyze-fixtures/Statement/DoNotUseGlobals.php]: Could you please fix PSR error here?

File: tests/analyze-fixtures/Statement/DoNotUseGlobals.php (hunk 2 above)
Comment (ddmler) [inline tests/analyze-fixtures/Statement/DoNotUseGlobals.php]: i meant you only need to remove it from the json here. the test runner ignores missing docblock and all others then.

Comment (micheleorselli) [inline tests/analyze-fixtures/Statement/DoNotUseGlobals.php]: ahh ok got it...

Comment (micheleorselli) [inline tests/analyze-fixtures/Statement/DoNotUseGlobals.php]: well it wasn't the case on my machine, I had to fix the test code

**Comment by ddmler** [comment]
Thanks @micheleorselli  :+1:  if you want to help more check our issues page, some easy things are marked with hacktoberfest.

**Comment by micheleorselli** [comment]
@ddmler thanks! will do!

**Pull request merged**